src/rng.rs

```diff
@@ -36,16 +66,75 @@ pub trait FiatShamirRng: RngCore {
     fn set_state(&mut self, new_state: Self::State);
 }
 
+#[derive(Default)]
+/// Implementation of `FiatShamirRngSeed` for `FiatShamirChaChaRng`.
+/// Output type of the seed is a byte array.
+/// To avoid cross-protocol attacks (see https://github.com/HorizenLabs/poly-commit/issues/22),
+/// the number of elements of the seed and their individual length will be part of the seed itself.
+pub struct FiatShamirChaChaRngSeed {
+    num_elements:  u64,
+    elements_len:  Vec<u64>,
+    seed_bytes:    Vec<u8>
+}
+
+impl FiatShamirRngSeed for FiatShamirChaChaRngSeed {
+    const MAX_NUM_ELEMENTS: usize = u64::MAX as usize;
```

Comment:
@UlrichHaboeck75 at this point this implementation is not circuit friendly and is not supposed to be like that, so we can avoid limiting MAX_NUM_ELEMENTS and MAX_ELEMENT_LEN thinking about the future circuit-friendly implementation

Reply:
I am not sure if I understand. In any case, we need to bound ``MAX_NUM_ELEMENTS`` and ``MAX_ELEMENT_LEN`` by the maximum value an u64 can carry. Otherwise our serialization implementation (add_bytes) is not one-to-one.

Reply:
It's not important right now, I was referring to the future, when we will have a Poseidon-based sponge and the need to pack these bytes in as little field elements as possible

Reply:
Let us write "we set it to the maximum, as chacha  is not meant to be circuit-friendly)."